src/System.Linq.Expressions/tests/ReadOnlyCollectionBuilderTests.cs

```diff
@@ -0,0 +1,914 @@
+            Assert.Equal(0, rocb.Count);
+        }
+
+        [Fact]
+        public void ReadOnlyCollectionBuilder_Capacity2()
+        {
+            var rocb = new ReadOnlyCollectionBuilder<int>(new[] { 1 });
+
+            Assert.Equal(1, rocb.Capacity);
+            Assert.Equal(1, rocb.Count);
+
+            Assert.Throws<ArgumentOutOfRangeException>("value", () => { rocb.Capacity = 0; });
+            Assert.Throws<ArgumentOutOfRangeException>("value", () => { rocb.Capacity = -1; });
+
+            rocb.Capacity = 1;
+
+            Assert.Equal(1, rocb.Capacity);
+            Assert.Equal(1, rocb.Count);
+
+            rocb.Capacity = 2;
+
+            Assert.Equal(2, rocb.Capacity);
+            Assert.Equal(1, rocb.Count);
+        }
+
+        [Fact]
+        public void ReadOnlyCollectionBuilder_Capacity3()
+        {
+            var rocb = new ReadOnlyCollectionBuilder<int>(new[] { 1 });
+
+            Assert.Equal(1, rocb.Capacity);
+            Assert.Equal(1, rocb.Count);
+
+            rocb.RemoveAt(0);
+
+            Assert.Equal(1, rocb.Capacity);
+            Assert.Equal(0, rocb.Count);
+
+            rocb.Capacity = 0;
+
+            Assert.Equal(0, rocb.Capacity);
+            Assert.Equal(0, rocb.Count);
+        }
+
+        [Fact]
+        public void ReadOnlyCollectionBuilder_IList_IsReadOnly()
+        {
+            IList rocb = new ReadOnlyCollectionBuilder<int>();
+
+            Assert.False(rocb.IsReadOnly);
+        }
+
+        [Fact]
+        public void ReadOnlyCollectionBuilder_ICollectionOfT_IsReadOnly()
+        {
+            ICollection<int> rocb = new ReadOnlyCollectionBuilder<int>();
+
+            Assert.False(rocb.IsReadOnly);
+        }
+
+        [Fact]
+        public void ReadOnlyCollectionBuilder_IList_IsFixedSize()
+        {
+            IList rocb = new ReadOnlyCollectionBuilder<int>();
+
+            Assert.False(rocb.IsFixedSize);
+        }
+
+        [Fact]
+        public void ReadOnlyCollectionBuilder_IList_IsSynchronized()
+        {
+            ICollection rocb = new ReadOnlyCollectionBuilder<int>();
+
+            Assert.False(rocb.IsSynchronized);
+        }
+
+        [Fact]
+        public void ReadOnlyCollectionBuilder_ICollection_SyncRoot()
+        {
+            ICollection rocb = new ReadOnlyCollectionBuilder<int>();
+
+            object root1 = rocb.SyncRoot;
+            Assert.NotNull(root1);
+
+            object root2 = rocb.SyncRoot;
+            Assert.Same(root1, root2);
+        }
+
+        [Fact]
+        public void ReadOnlyCollectionBuilder_IndexOf()
+        {
+            var rocb = new ReadOnlyCollectionBuilder<int>(new[] { 1, 2, 2, 3 });
+
+            Assert.Equal(4, rocb.Count);
+
+            Assert.Equal(0, rocb.IndexOf(1));
+            Assert.Equal(1, rocb.IndexOf(2));
+            Assert.Equal(3, rocb.IndexOf(3));
+
+            Assert.True(rocb.IndexOf(0) < 0);
+            Assert.True(rocb.IndexOf(4) < 0);
```

Comment:
These and others like them can use Assert.InRange instead, e.g.
```C#
Assert.InRange(rocb.IndexOf(4), int.MinValue, -1);
```
That way if the assert fails, the failure will include what the value actually was.